torch/csrc/lazy/core/shape_inference.cpp

```diff
@@ -418,6 +422,37 @@ std::vector<Shape> compute_shape_embedding_dense_backward(
       Shape(grad_output.scalar_type(), {num_weights, grad_output.size(-1)})};
 }
 
+std::vector<Shape> compute_shape_expand(const at::Tensor & self, at::IntArrayRef size, bool implicit) {
+  CHECK_GE(size.size(), self.dim());
+  int64_t num_new_dimensions = size.size() - self.dim();
+  std::vector<int64_t> padded_self(num_new_dimensions, 0);
+  padded_self.insert(padded_self.end(), self.sizes().begin(),
+                     self.sizes().end());
+  std::vector<int64_t> target_size(size.size());
+  for (const auto idx : c10::irange(size.size())) {
+    target_size[idx] = size[idx] == -1 ? padded_self[idx] : size[idx];
+  }
+  return {Shape(self.scalar_type(), target_size)};
+}
+
+std::vector<Shape> compute_shape_expand(const at::Tensor & self, c10::SymIntArrayRef size, bool implicit) {
+  CHECK_GE(size.size(), self.dim());
+  std::vector<c10::SymInt> _sizes = ToVector<c10::SymInt>(size);
+  int64_t num_new_dimensions = _sizes.size() - self.dim();
+  std::vector<int64_t> padded_self(num_new_dimensions, 0);
+  padded_self.insert(padded_self.end(), self.sizes().begin(),
+                     self.sizes().end());
+  std::vector<int64_t> target_size(_sizes.size());
+  for (const auto idx : c10::irange(_sizes.size())) {
+    std::shared_ptr<c10::SymbolicIntNode> symbolicIntNode = _sizes[idx].toSymbolicIntNode();
+    auto lazySymIntNode = std::dynamic_pointer_cast<torch::lazy::SymbolicIntNode>(symbolicIntNode);
+    auto size_node = lazySymIntNode->node_;
+    auto static_value = std::dynamic_pointer_cast<torch::lazy::DimensionNode>(size_node)->getStaticValue();
+    target_size[idx] = static_value == -1 ? padded_self[idx] : static_value;
```

Comment:
@miladm I thought we decided to keep `-1` as a plain integer? @suo added support for this case. 
I think your check should say: 
`_sizes[idx].is_symbolic()` then do 

```
 auto lazySymIntNode = std::dynamic_pointer_cast<torch::lazy::SymbolicIntNode>(symbolicIntNode);
    auto size_node = lazySymIntNode->node_;
    auto static_value = std::dynamic_pointer_cast<torch::lazy::DimensionNode>(size_node)->getStaticValue();
   ```
   
   otherwise just do `_sizes[idx].data()` .

Reply:
Addressed.